mam/v2/mss.c

```diff
@@ -393,7 +397,13 @@ void mss_sign(mss_t *mss, trits_t H, trits_t sig) {
     wots_gen_sk3(mss->wots, mss->prng, mss->nonce1, mss->nonce2, Ni);
   }
 
-  wots_sign(mss->wots, H, trits_take(sig, MAM2_WOTS_SIG_SIZE));
+  flex_trits_to_trits(sig.p + sig.d, MAM2_WOTS_SK_SIZE, mss->wots->sk,
+                      MAM2_WOTS_SK_SIZE, MAM2_WOTS_SK_SIZE);
+  TRIT_ARRAY_MAKE_FROM_RAW(hash_array, MAM2_WOTS_HASH_SIZE, hash.p + hash.d);
+  TRIT_ARRAY_MAKE_FROM_RAW(sk_sig_array, MAM2_WOTS_SIG_SIZE, sig.p + sig.d);
```

Comment:
Can we use TRIT_ARRAY_DECLARE instead ?

Reply:
sig is initialized with `sk` but for hash, yes

Reply:
Oops, missed that